lib/3scale_toolbox/commands/copy_command/copy_service.rb

```diff
@@ -0,0 +1,214 @@
+require 'cri'
+require '3scale_toolbox/base_command'
+
+module ThreeScaleToolbox
+  module Commands
+    module CopyCommand
+      module CopyServiceSubcommand
+        extend ThreeScaleToolbox::Command
+        def self.command
+          Cri::Command.define do
+            name        'service'
+            usage       'service [options] <service_id>'
+            summary     '3scale CLI copy service'
+            description '3scale CLI copy service tool'
+
+            flag :h, :help, 'show help for this command' do |_, cmd|
+              puts cmd.help
+              exit 0
+            end
+
+            required  :s, :source, 'Source'
+            required  :d, :destination, 'Destination'
+            required  :t, 'target_system_name', 'Target system name'
+
+            run do |opts, args, _|
+              CopyServiceSubcommand.run opts, args
+            end
+          end
+        end
+
+        def self.run(opts, args)
```

Comment:
Could (sub)commands be classes too? I guess so, right?

It does not affect the plugin system at all, but I think it would be nice as an example to actually instantiate the class with parameters and then run it.

Reply:
Sure. Just change `module FooCommand` for `class FooCommand` and example should work. But instantiate the class is not going to work as it is know. I will think about it. Initially modules are used because it did not make sense to create "objects" of commands.

Reply:
The "command" can still rely on class level `.run`, but that one would do `.new(opts, args).call` to actually instantiate the class and call it. 
That will allow "commands" to have state per execution without messing global/class variables.

Reply:
Something like this?

```
$ cat lib/3scale_toolbox_plugin.rb
require '3scale_toolbox/cli'

class FooCommand
  extend ThreeScaleToolbox::Command

  def self.command
    Cri::Command.define do
      name        'foo'
      usage       'foo [options]'
      summary     '3scale copy foo'
      description '3scale copy foo subcommand'
      flag :h, :help, 'show help for this command' do |_, cmd|
        puts cmd.help
        exit 0
      end
      run do |opts, args, _|
        FooCommand.new(opts, args).call
      end
    end
  end

  def initialize(opts, args)
    @opts = opts
    @args = args
  end

  def call
    puts "foo done"
  end
end

ThreeScaleToolbox::Commands::CopyCommand.add_subcommand(FooCommand)
```
Then running
```
$ RUBYOPT=-Ilib 3scale copy foo
foo done
```

This example works perfectly

Reply:
Yep. Something like that. 👍 🥇 

I think we should "promote" this kind of interface instead of class level one, because it is harder to make mistakes (like instance level variables in class methods).

But that does not need to be concern of this PR.